src/scripts/worker/worker.js

```diff
@@ -0,0 +1,101 @@
+(function(tiy){
+  'use strict';
+  const hostName = "com.theironyard.newlinecli.hw";
+  let connection;
+  let pendingPromises = [];
+
+  tiy.worker.connect = connect;
+  tiy.worker.onNativeMessage = onNativeMessage;
```

Comment:
Do we need this on the namespace? I don't think we use this in the non-worker JS, do we? Same for the `onDisconnected` handler and `pendingPromises` below...